cmd/kube-controller-manager/app/plugins.go

```diff
@@ -91,12 +89,14 @@ func NewVolumeProvisioner(cloud cloudprovider.Interface, config componentconfig.
 	switch {
 	case cloud == nil && config.EnableHostPathProvisioning:
 		return getProvisionablePluginFromVolumePlugins(host_path.ProbeVolumePlugins(volume.VolumeConfig{}))
-	case cloud != nil && aws.ProviderName == cloud.ProviderName():
-		return getProvisionablePluginFromVolumePlugins(aws_ebs.ProbeVolumePlugins())
-	case cloud != nil && gce.ProviderName == cloud.ProviderName():
-		return getProvisionablePluginFromVolumePlugins(gce_pd.ProbeVolumePlugins())
-	case cloud != nil && openstack.ProviderName == cloud.ProviderName():
-		return getProvisionablePluginFromVolumePlugins(cinder.ProbeVolumePlugins())
+	case config.EnableNetworkStorageProvisioning:
```

Comment:
Problem with this approach is that `--enable-network-storage-provisioner` is now just for glusterfs. How are you going to decide if gluster *or* ceph should be initialized here? Current provisioner logic allows only one provisioner, ceph or gluster (or the cloud one, say AWS EBS).